tests/test_endpoints.py

```diff
@@ -16,64 +18,66 @@ def test_cli(loop, app, sanic_client):
     return loop.run_until_complete(sanic_client(app))
 
 
-async def test_index(test_cli):
-    resp = await test_cli.get('/')
-    assert resp.status == 200
+@pytest.fixture
+def generateData(request):
+    # adding 5 Language documents
+    insertDummyLanguages()
+
+    def fin():
+        removeCollections()
+
+    request.addfinalizer(fin)
 
 
-async def test_getLanguages_with_no_query_param(test_cli):
+async def test_getLanguages_with_noParam_shouldPass(test_cli, generateData):
     resp = await test_cli.get('/languages')
     resp_json = await resp.json()
-    assert resp_json["total_results"] == 0
+    assert resp_json["total_results"] == 5
+    assert len(resp_json["results"]) == 5
     assert resp_json["page"] == 1
     assert resp.status == 200
 
 
-async def test_getLanguages_with_more_query_param(test_cli):
-    params = {'page': 2, 'type': 'local', 'limit': 50}
+async def test_getLanguages_with_moreThanExpectedParam_shouldShowErrorMsg(test_cli, generateData):
+    params = {'page': 2, 'type': 'local', 'no_valid_param': 50}
     resp = await test_cli.get('/languages', data=json.dumps(params))
     resp_json = await resp.json()
     assert resp_json["message"] == "Expecting less than 3 arguments"
     assert resp.status == 400
 
 
-async def test_getLanguages_with_valid_query_param(test_cli):
+async def test_getLanguages_with_bothExpectedParam_shouldPass(test_cli, generateData):
     params = {'page': 2, 'type': 'local'}
     resp = await test_cli.get('/languages', data=json.dumps(params))
     resp_json = await resp.json()
-    assert resp_json["total_results"] == 0
+    assert resp_json["total_results"] == 3
+    assert len(resp_json["results"]) == 0
```

Comment:
what is the difference between total_resuls and results? Why are you not checking for the data? I am of opinion that total_results should not be part of the response but your call.